packages/app/src/cli/services/app-logs/write-app-logs.ts

```diff
@@ -0,0 +1,35 @@
+import {AppEventData} from './poll-app-logs.js'
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {writeLog, getLogsDir} from '@shopify/cli-kit/node/logs'
+
+import {Writable} from 'stream'
+
+export const writeAppLogsToFile = async ({
+  appLog,
+  apiKey,
+  stdout,
+}: {
+  appLog: AppEventData
+  apiKey: string
+  stdout: Writable
+}) => {
+  const fileName = `app_logs_${appLog.log_timestamp}.json`
+  const path = joinPath(apiKey, fileName)
+  const fullOutputPath = joinPath(getLogsDir, path)
+  const toSaveData = {
+    ...appLog,
+  }
+
+  try {
+    const parsedPayload = JSON.parse(appLog.payload)
+    toSaveData.payload = parsedPayload
```

Comment:
Nit: Instead of mutating the object here, we could move the initialization down here:

```typescript
const toSaveData = {
  ...appLog,
  payload: JSON.parse(appLoad.payload),
}
```